biogeophys/FatesPlantRespPhotosynthMod.F90

```diff
@@ -1790,6 +1790,8 @@ subroutine quadratic_f (a, b, c, r1, r2)
   ! !REVISION HISTORY:
   ! 4/5/10: Adapted from /home/bonan/ecm/psn/An_gs_iterative.f90 by Keith Oleson
   ! 7/23/16: Copied over from CLM by Ryan Knox
+  ! 12/30/23: Instead of issuing errors when a=0, solve the trivial cases too.
+  !           Check determinant sign, and stop the run when it is negative.
```

Comment:
Add an explicit reference to the Numerical Solutions book in case the person reading the code isn't familiar with it.

Reply:
Thanks for revising it, @glemieux! There was already a reference to the book in the comments, but for an earlier edition. I updated with the full reference and link.